src/frontends/tensorflow_common/src/op/pad.cpp

```diff
@@ -82,10 +83,39 @@ OutputVector translate_pad_op(const NodeContext& node) {
 }
 
 OutputVector translate_padv2_op(const NodeContext& node) {
-    default_op_checks(node, 3, {"PadV2"});
+    default_op_checks(node, 3, {"PadV2"}, true);
     auto input = node.get_input(0);
     auto paddings = node.get_input(1);
     auto constant_value = node.get_input(2);
+    if (auto complex_type_mark = as_type_ptr<ComplexTypeMark>(input.get_node_shared_ptr())) {
+        input = complex_type_mark->input_value(0);
+        element::Type complex_part_type = complex_type_mark->get_complex_part_type();
+
+        auto gather_index_real = make_shared<v0::Constant>(element::i32, Shape{}, 0);
+        auto gather_index_imag = make_shared<v0::Constant>(element::i32, Shape{}, 1);
+        auto minus_one = make_shared<v0::Constant>(element::i32, Shape{1}, -1);
+        auto x_real = make_shared<v8::Gather>(input, gather_index_real, minus_one)->output(0);
+        auto x_imag = make_shared<v8::Gather>(input, gather_index_imag, minus_one)->output(0);
+
+        auto constant_complex_type_mark = as_type_ptr<ComplexTypeMark>(constant_value.get_node_shared_ptr());
+        auto constant_input = constant_complex_type_mark->input_value(0);
+        auto constant = ov::util::get_constant_from_source(constant_input);
+        std::vector<float> constant_vector = constant->cast_vector<float>();
+        auto constant_value_real = create_same_type_const_scalar<float>(input, constant_vector[0]);
+        auto constant_value_imag = create_same_type_const_scalar<float>(input, constant_vector[1]);
```

Comment:
Let us do not do this because `constant_input ` not neccesarily to be `Constant` node and it can be under `ShapeOf` sub-graph. Let us also compute real and imag parts using `Gather` as above. This way we do not need to re-create `Constant` node

Reply:
@rkazants Thank you for your comments! I have incorporated your suggestions. Can you please review this PR again? 

Also, I'm not clear on the sentence "it can be under ShapeOf sub-graph" in your comments. Could you please explain it?